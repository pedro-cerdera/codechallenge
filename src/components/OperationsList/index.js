import OperationsList from './OperationsList';

export {OperationsList};
