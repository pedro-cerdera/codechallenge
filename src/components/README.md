
# Components

Here we have some components

- [OperationsList](/src/components/OperationsList/README.md)
- [Account](/src/components/Account/README.md)
