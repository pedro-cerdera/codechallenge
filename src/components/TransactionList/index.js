import TransactionList from './TransacitonList';

export {TransactionList};
