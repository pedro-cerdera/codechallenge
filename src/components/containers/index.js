export * from './SafeAreaContainer';
export * from './CardContainer';
export * from './DefaultScapingContainer';
