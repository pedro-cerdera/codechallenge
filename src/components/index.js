export * from './containers';
export * from './Account';
export * from './Icon';
export * from './OperationsList';
