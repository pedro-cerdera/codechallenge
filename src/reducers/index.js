import {TransactionReducer, Actions as TranscationActions} from './transaction';

export default {TransactionReducer};

export const Actions = {Transaction: TranscationActions};
