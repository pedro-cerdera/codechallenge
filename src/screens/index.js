export * from "./Home";
export * from "./NewTransaction";
export * from "./Transactions";