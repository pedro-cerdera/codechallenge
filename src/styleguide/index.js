import colors from './Colors';
import typografy from './Typografy';

export default {colors, typografy};
