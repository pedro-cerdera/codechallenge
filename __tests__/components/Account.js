import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';

import {Account} from '../../src/components';

const mockPress = jest.fn();
const mockMaskMoney = jest.fn();

jest.mock('../../src/components/Icon/Icon', () => 'Icon');
jest.mock('../../src/helpers/Text', () => ({
  maskMoney: (v) => mockMaskMoney(v),
}));

describe('Testing Account', () => {
  test('Render all account parts correctly', () => {
    render(<Account ammount={100} onBottomPress={mockPress} />);
    expect(mockMaskMoney).toHaveBeenCalledWith(100);
    expect(mockMaskMoney).toHaveBeenCalledTimes(1);
    expect(mockPress).not.toHaveBeenCalled();
  });

  test('Test click operation item', () => {
    const {getByTestId} = render(
      <Account ammount={200} onBottomPress={mockPress} />,
    );

    fireEvent.press(getByTestId('bottom-button'));
    expect(mockPress).toHaveBeenCalledTimes(1);
  });

  test('renders correctly', () => {
    const tree = render(
      <Account ammount={100} onBottomPress={mockPress} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
