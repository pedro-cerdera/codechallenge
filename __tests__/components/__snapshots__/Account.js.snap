// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Account renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 5,
        "elevation": 1,
        "maxWidth": 500,
        "shadowColor": "gray",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
        "width": "100%",
      },
      Object {
        "flex": 1,
        "justifyContent": "space-between",
        "maxHeight": 350,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 30,
        "paddingTop": 30,
      }
    }
  >
    <Icon
      color="black"
      name="fa5-wallet"
      rightSpacing={true}
      size={24}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#597183",
            "fontFamily": "Poppins-Regular",
            "fontSize": 18,
          },
          undefined,
        ]
      }
    >
      Conta
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#597183",
            "fontFamily": "Poppins-Regular",
            "fontSize": 18,
          },
          undefined,
        ]
      }
    >
      Saldo disponível
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#597183",
            "fontFamily": "Poppins-Regular",
            "fontSize": 20,
          },
          undefined,
        ]
      }
      testID="ammount-value"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000009",
        "flexDirection": "row",
        "opacity": 1,
        "padding": 30,
      }
    }
    testID="bottom-button"
  >
    <Icon
      color="black"
      name="fa5-list"
      rightSpacing={true}
      size={18}
    />
    <Text
      style={
        Array [
          Object {
            "color": "#597183",
            "fontFamily": "Poppins-Regular",
            "fontSize": 14,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      Veja suas transações
    </Text>
    <Icon
      color="black"
      name="fa-angle-right"
      size={24}
    />
  </View>
</View>
`;
